replica/replica.go

```diff
@@ -903,16 +960,16 @@ func (r *Replica) createDisk(name string, userCreated bool, created string) erro
 
 	f, newHeadDisk, err := r.createNewHead(oldHead, newSnapName, created)
 	if err != nil {
+		r.doCleanup(newHeadDisk.Name)
```

Comment:
these changes are not related to this PR

Reply:
This is to handle the corner case of cleaning up the leftovers due to error.